packages/astro/src/assets/build/remote.ts

```diff
@@ -19,6 +19,40 @@ export async function loadRemoteImage(src: string) {
 	return {
 		data: Buffer.from(await res.arrayBuffer()),
 		expires: Date.now() + expires,
+		etag: res.headers.get('Etag'),
+	};
+}
+
+export async function revalidateRemoteImage(src: string, etag: string) {
+	const req = new Request(src, { headers: { 'If-None-Match': etag } });
+	const res = await fetch(req);
+
+	if (!res.ok && res.status != 304) {
+		throw new Error(
+			`Failed to revalidate cached remote image ${src}. The request did not return a 200 OK / 304 NOT MODIFIED response. (received ${res.status}))`,
```

Comment:
This error isn't actionable. If a user sees that, they don't know what to do in order to fix it. We should provide a better error that gives the user an action.

Reply:
I agree that this is a bad error, however this is handled the same way in the existing `loadRemoteImage` function I based the `revalidateRemoteImage` function off of, which provides similarly un-actionable errors. 

I can potentially handle common cases like 404 with custom messages although there are many status codes where it would be difficult to give any actionable advice, like 5xx errors.

Reply:
Yeah a warning would be a better solution. I've added the status text (e.g. NOT FOUND/FORBIDDEN) to the error, which should make it more understandable without needing to look up the status code first. 

I've added a try catch block in `generate.ts` to handle this error and errors from Request itself, which should fall back to using the stale cache and print a warning in the log.

Example warning:
```
[WARN] An error was encountered while revalidating a cached remote asset. Proceeding with stale cache. Error: Failed to revalidate cached remote image https://dam.cms.io/test.jpg. The request did not return a 200 OK / 304 NOT MODIFIED response. (received 403 Forbidden)
  ▶ /_astro/test_Z2qHSPG.webp (reused cache entry) (+295ms) (31/33)
```